### PR TITLE
Codestyle: Add typecheck and unused linters

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi @torkelo, @bergquist, @xlson,

related to #10381, I fixed the **typecheck** issues, and also added this linter, and **[unused](https://github.com/dominikh/go-tools/tree/master/cmd/unused)** to the CI.

### CODE REVIEW

File: scripts/build/publish_test.go [hunk 1]
```diff
@@ -1,110 +0,0 @@
-package main
```
Comment (bergquist) [inline scripts/build/publish_test.go]: Removing this file seems weird... Its something we want to keep.

Comment (mjtrangoni) [reply scripts/build/publish_test.go]: @bergquist well, `scripts/build/publish.go` was deleted [here](https://github.com/grafana/grafana/commit/55f2495afa043b67f9fc567d59c58eccc5d767c8). Why do we want to keep it?

Comment (bergquist) [reply scripts/build/publish_test.go]: Oh sorry. didn't catch that.

**Review by bergquist — APPROVED** [review]
LGTM

Thank you for contributing!

**Pull request merged**